config/v1/types_infrastructure.go

```diff
@@ -87,6 +92,21 @@ const (
 	VSpherePlatformType PlatformType = "VSphere"
 )
 
+// PlatformStatus holds the current status specific to the underlying infrastructure provider
+// of the current cluster. Since these are used at status-level for the underlying cluster, it
+// is supposed that only one of the status structs is set.
+type PlatformStatus struct {
```

Comment:
This needs a discriminator field which indicates which one of the many values is going to be set.  CRD discriminators are assumed to be in the same struct as the subfields I think.  @sttts

Reply:
in the CRD it must be in the same JSON object (which in case of struct embedded might differ from "same struct"). Generators like crd-gen from controller-tools will probably require the same struct.

Reply:
So bring https://github.com/openshift/api/blob/a1f3bdc32a84ec053166e5da150e0c504e3a14e6/config/v1/types_infrastructure.go#L48 down into here?  And deprecate the old property?

Reply:
...deprecate for removal in v2 in four years ;)

Reply:
and that's not only a restriction for CRDs, but will be for native types as well.

Reply:
> ...deprecate for removal in v2 in four years ;)

I think you're left with no other choice.

Reply:
I'm not sure if native API adds discriminators like
https://godoc.org/k8s.io/api/core/v1#VolumeSource

Reply:
Ok, pushed the discriminator down into `platformStatus` deprecating the old property with a1f3bdc3 -> 81ef6b42.